File: algo-anagrams-i/js/characterMatchSpec.js
var ana = require("./characterMatch");

// Part 1
console.log(ana.isCharacterMatch('charm', 'march') === true);
console.log(ana.isCharacterMatch('zach', 'attack') === false);
console.log(ana.isCharacterMatch('CharM', 'mARcH') === true);
console.log(ana.isCharacterMatch('abcde2', 'c2abed') === true);

console.log("This test is for the challenge quesiton");
console.log(ana.isCharacterMatch('Anna Madrigal', 'A man and a girl') === true);


// Part 2
listOfWords = ["threads", "trashed", "hardest", "hatreds", "hounds"];

console.log(ana.anagramsFor("threads", listOfWords).length == 4);
console.log(ana.anagramsFor("threads", listOfWords)[0] == "threads");
console.log(ana.anagramsFor("threads", listOfWords)[1] == "trashed");
console.log(ana.anagramsFor("threads", listOfWords)[2] == "hardest");
console.log(ana.anagramsFor("threads", listOfWords)[3] == "hatreds");

console.log(ana.anagramsFor("apple", listOfWords).length == 0);

File: Algorithms/algo-anagrams-I/python/character_match_spec.py
from character_match import is_character_match, anagrams_for

# Part 1
print(is_character_match('charm', 'march') == True)
print(is_character_match('zach', 'attack') == False)
print(is_character_match('CharM', 'mARcH') == True)
print(is_character_match('abcde2', 'c2abed') == True)

print("This test is for the challenge quesiton")
print(is_character_match('Anna Madrigal', 'A man and a girl') == True)

# Part 2

list_of_words = ["threads", "trashed", "hardest", "hatreds", "hounds"]
list_of_words2 = ["threads", "tra   she  d", "har.d,e/st", "hat8red+s"]
print(anagrams_for("threads", list_of_words) == ["threads", "trashed", "hardest", "hatreds"])
print(anagrams_for("threads", list_of_words2) == ["threads", "trashed"])
print(anagrams_for("apple", list_of_words) == [])
